daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -313,12 +315,38 @@ object Hash {
   def safeHashContractKey(templateId: Ref.Identifier, key: Value[Nothing]): Hash =
     assertHashContractKey(templateId, key)
 
-  def hashContractKey(
+  private def comittedCidToBytes(coid: ContractId): Bytes =
+    coid match {
+      case coid: ContractId.V1 => coid.toBytes
+      case _: ContractId.V0 => error("hashing of Contract ID V0 not supported")
+    }
+
+  // This function assumes that key is well typed, i.e. :
+  // 1 - `templateId` is the identifier for a template with a key of type τ
+  // 2 - `key` is a value of type τ
+  @throws[HashingError]
+  def assertHashContractKey(
```

Comment:
Minor: Given that this method returns an `Either`, I'm somewhat surprised that it has `assert` in its name.